cranelift/codegen/src/machinst/abi.rs

```diff
@@ -927,6 +902,23 @@ impl SigSet {
 
         clobbers
     }
+
+    /// Get the number of arguments expected.
+    pub fn num_args(&self, sig: Sig) -> usize {
+        let sig_data = &self.sigs[sig];
+        let num = if sig_data.stack_ret_arg.is_some() {
+            self.args(sig).len() - 1
+        } else {
+            self.args(sig).len()
+        };
+
+        usize::from(num)
```

Comment:
This version of `num_args` would be slightly simpler if it looked more like the old definition of `Caller::num_args`:

```suggestion
        let len = self.args(sig).len();
        if self.sigs[sig].stack_ret_arg.is_some() {
            len - 1
        } else {
            len
        }
```